test/unit/message_test.rb

```diff
@@ -141,8 +141,8 @@ class MessageTest < ActiveSupport::TestCase
     assert_equal 1, message.errors[:sender_id].size
   end
 
-  test 'A message not require a subject' do
-    message = FactoryBot.build(:message, subject: nil, sender: FactoryBot.create(:simple_account))
+  test 'A message require a subject' do
+    message = FactoryBot.build(:message, subject: "I am subject", sender: FactoryBot.create(:simple_account))
```

Comment:
This is testing that an object of `Message` with subject AND sender (and maybe something else from the Factory) is valid, which's more than the description lets us know.
Maybe a shorter way to test this is to instantiate a `Message` without a `subject` and check that it's invalid AND that there's an error message for `subject`.

Reply:
I think this test failed because there were no subject and if we have added validation of required filed so I think we must enter it in test also so if in future we change something atleast things reflect what failed or not

Reply:
I have added one more test which cover your explained scenario as well